### PR TITLE
Added form node

### PR DESCRIPTION
**Pull request by hrhrprasath**
Helps to collect a bunch of information from the user

### CODE REVIEW

File: src/components/ui-component/ui-component-ctrl.js [hunk 1]
```diff
@@ -96,6 +96,14 @@ angular.module('ui').controller('uiComponentController', ['$scope', 'UiEvents',
                         };
                     }
```
Comment (dceejay) [inline src/components/ui-component/ui-component-ctrl.js]: do you need a break; here ?

File: src/components/ui-component/templates/form.css [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	height :auto !important;
+}
+
+.nr-dashboard-form-button {
+	width:50%;
+	float:left;
+  background-color:  #0094ce !important;
+	color: rgba(255, 255, 255, 0.87) !important;
+}
+.nr-dashboard-theme-dark  .nr-dashboard-form-button {
```
Comment (dceejay) [inline src/components/ui-component/templates/form.css]: All the colour css parts should reside in the two main theme files - dark.css and light.css.

**Comment by dceejay** [comment]
Hi,
see what you are trying to do... - I have added some comments to the files above.

Also the alignment is going to need some work
image
image

As are the colours... (which as per comments above - the css for the colours should be in the theme/dark.css and theme/light.css files
Also in here 
image
the Value should probably be called Name  -as it's the name part of the object that is finally created.

**Comment by dceejay** [comment]
Hi - getting there...
in the edit panel - the left hand drag bars don;t seem to let me rearrange the order...
image

There are still some sizing issues if multiple items on the form... It should try to fit them all if the size is set to auto.
image

If I enter any text in the text box widget and hit enter it goes red - same with password box
image

The right arrow dropdown on the date widget is blank (hard to see on dark theme)
image

Switch in off position is white (vs the grey we have for standalone switch)
image

The text input is also styled differently from our default
yours 
image
image
  vs ours 
image 
image
which has the side effect of making it taller as well - (which will also make the overall height calculation more complicated....

In fact the other widgets are also slightly differently styled. (text colour)

But still - much better than it was :-)

I do have to ask the question... why do we need this ? admittedly we don't have a date picker - but could we not build a form out of parts we already have ? - accumulating the field values in a function node until it sees the submit button action ?

**Comment by dceejay** [comment]
Also the date picker - you are using the standard html5 entry field with a date type. This is not yet supported on firefox or safari... (this is quite a good site to get a heads up on this sort of thing) - http://caniuse.com/#feat=forms

**Comment by hrhrprasath** [comment]
I am not using native html5 date picker,I am using mdCalendar directive of Angular material 
https://material.angularjs.org/latest/api/directive/mdCalendar

**Comment by dceejay** [comment]
Hi,
much better.
when I hit submit it clears the form then complains in red that I need to fill in the fields. Is there any way it can not re-validate the empty form after submit ?

image

Also I think the vertical spacing is still a bit large - the label "floats" a bit too far away from the field... opinion please @knolleary ? (but the form fits into the space correctly now)
image

Date picker - much better - works across  Chrome, FF and Safari - Thanks

**Comment by dceejay** [comment]
While the date picker is now much better - it does format date as US style  - which will be the first thing people will ask to change... Maybe still better to drop for now.
image

**Comment by hrhrprasath** [comment]
@dceejay Is there any way it can not re-validate the empty form after submit ?
No we are using ng- required so whenever we clear the data it automatically validated the form.

**Comment by dceejay** [comment]
Well - it's ugly... does this help ? 
5 seconds on google... 
http://stackoverflow.com/questions/17452247/validate-form-field-only-on-submit-or-user-input

**Comment by hrhrprasath** [comment]
@dceejay fixed that submit issue

**Comment by dceejay** [comment]
That is sooooo much better thanks.
So - final question... do you really need the date picker for your scenario ? 
While it looks good and works cross browser, the US only formatting will not be readily acceptable to most users (as mm/dd/yyyy is really only in 2 or 3 countries :-) - We don't yet have a strategy to allow customisation of this - and nor do we wish to rush to force one in in the short timescale you possibly need.

So would it be ok to comment out the date picker option for now ?  Or does your scenario really need it on day one ?

Finally - can you remove the dist files from the PR - as they conflict with the existing ones - when we can just rebuild them anyway.

**Comment by dceejay** [comment]
Merged manually, due to conflicts.

**Pull request closed without merging**